src/tests/sqlsmith/src/sql_gen/func.rs

```diff
@@ -649,21 +652,28 @@ impl<R: Rng> SqlGenerator<'_, R> {
                     "last",
                     "nth_value",
                 ];
-                let name = name[self.rng.gen_range(0..=6)];
+                let idx = self.rng.gen_range(0..names.len());
+                let name = names[idx];
                 let args_type = if name == "lag" || name == "lead" {
                     vec![ty; 3]
                 } else if name == "nth_value" {
                     vec![ty, DataType::Number(NumberDataType::UInt64)]
                 } else {
                     vec![ty]
                 };
+                let is_ignore_nulls = idx > 1;
+                let window = self.gen_window(is_ignore_nulls);
                 self.gen_func(name.to_string(), vec![], args_type, window, None)
             }
         }
     }
 
-    fn gen_window(&mut self) -> Option<WindowDesc> {
-        let ignore_nulls = Some(self.rng.gen_bool(0.2));
+    fn gen_window(&mut self, is_ignore_nulls: bool) -> Option<WindowDesc> {
+        let ignore_nulls = if is_ignore_nulls {
```

Comment:
consider add some check? Now only last/first_value and "last"/"first" support ignore_null